sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -224,7 +223,7 @@ class UnivocityParser(
           case NonFatal(e) =>
             // If fails to parse, then tries the way used in 2.0 and 1.x for backwards
             // compatibility if enabled.
-            if (!enableParsingFallbackForTimestampType) {
+            if (!enableParsingFallbackForDateType) {
```

Comment:
Thanks for fixing it, the values were swapped.

Reply:
@xiaonanyang-db Does it mean that we don't have test coverage for it? I am wondering how I managed to merge like this. I will open a follow-up PR to improve testing.

Reply:
That will be nice!

Reply:
After thinking about it more, let me submit a separate PR to fix it, this is a serious bug and I will fix it ASAP, we may need to port to other Spark branches.

Reply:
Opened https://github.com/apache/spark/pull/37942 to fix this. Apologies for the mistake.

Reply:
This has been fixed, could you rebase?